rbac/default-role-manager/role_manager.go

```diff
@@ -449,6 +451,11 @@ func NewDomainManager(maxHierarchyLevel int) *DomainManager {
 	return dm
 }
 
+func (dm *DomainManager) SetTenantManager(tm rbac.RoleManager) {
```

Comment:
Needs documentation on how to use this. Though at the same time, shouldn't the hierarchy be enabled by default?

What is base definition for Named Grouping Policies? If you set up a grouping policy for domains shouldn't they be hierarchical by default? @hsluoyz

Reply:
I don't quite understand what you are saying. This PR is to use the RoleManager to manage the hierarchy of the domain

Reply:
I was thinking out loud and more broadly how hierarchy should be enabled for the DomainManager. Not something that is necessary to solve in this pull request.